src/compiler/emitter.ts

```diff
@@ -1875,19 +1888,68 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                 }
                 else {
-                    write("(");
-                    emitCommaList(node.arguments);
-                    write(")");
+                    emitBracedEnclosedCommaList(node.arguments);
                 }
             }
-
-            function emitNewExpression(node: NewExpression) {
+            
+            /** Emit a new expression. 
+              * 
+              * Emit call helper:
+              * When we are emitting a call expression from a new expressions with spread syntax in ES5 
+              * we need an argument to help this function decide whether it should emit a pair of extra 
+              * braces "()" at the end. These extra braces are there to help the expression to not be 
+              * consumed by the new expression directly.
+              * 
+              *   Example:
+              * 
+              *     new Array(...arguments)();
+              *
+              *     Should be transpiled into ES5:
+              *
+              *     new (Array.bind.apply(Array, [void 0].concat(arguments)))()()
+              */
+            function emitNewExpression(node: NewExpression, emitCallHelper = false) {
                 write("new ");
-                emit(node.expression);
-                if (node.arguments) {
+
+                // Spread operator logic can be supported in new expressions in ES5 using a combination
+                // of Function.prototype.bind() and Function.prototype.apply().
+                //
+                //     Example:
+                //
+                //         var arguments = [1, 2, 3, 4, 5];
+                //         new Array(...arguments);
+                //
+                //         Could be transpiled into ES5:
+                //
+                //         var arguments = [1, 2, 3, 4, 5];
+                //         new (Array.bind.apply(Array, [void 0].concat(arguments)));
+                //
+                // `[void 0]` is the first argument which represents `thisArg` to the bind method above. 
+                // And `thisArg` will be set to the return value of the constructor when instantiated 
+                // with the new operator — regardless of any value we set `thisArg` to. Thus, we set it 
+                // to an undefined, `void 0`.
+                if (languageVersion === ScriptTarget.ES5 &&
+                    node.arguments &&
+                    hasSpreadElement(node.arguments)) {
+
                     write("(");
-                    emitCommaList(node.arguments);
-                    write(")");
+                    let target = emitCallTarget(node.expression);
+                    write(".bind.apply(");
+                    emit(target);
+                    write(", [void 0].concat(");
+                    emitListWithSpread(node.arguments, /*multiline*/false, /*trailingComma*/false);
+                    write(")))");
+                    if (emitCallHelper) {
+                        write("()");

```

Comment:
Oh, I see what you are doing. To simplify this even more, I would just always emit parentheses, regardless of whether the NewExpression is parented by a CallExpression. It's more idiomatic anyway.